### ISSUE TITLE
write points: write throughput too high. backoff and retry

### ISSUE THREAD

**Issue by brunohass** [influxdata/influxdb#4181]
Hi. We are an internet service provider. We have a CDN service using varnish, we generate statistics using varnishlog + python + influxdb module. We have as a customer some portals of local news. In peak times we even had over 50k written per second to the influxdb.

For the third time some customers have contacted us because they could not view the data in the dashboard. When I accessed the server over ssh and used the command-line tool (influx) performed a simple query and not getting any results. The influxdb seems that crashed.

Looking at the logs, had * many * messages such as the following:

```
write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
```

Our script that inserts the data into influxdb was unable to write. The problem was only resolved after a restart on influxdb. Even the restart took around 5 minutes.

Any idea? Below some information the moment it was happening the problem and the configuration file used.

Our hardware:

Dell PowerEdge R730
2x Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz
321GB RAM DDR4
16x Dell SSD (RAID6)

Slackware Linux 14.1 (using pre-built packages for debian)

Currently this server just runs influxdb and apache + wsgi + django.

Thanks in advance. And I congratulate you for your excellent software!

```
$ influx
Connected to http://localhost:8086 version 0.9.4.1
InfluxDB shell 0.9.4.1
> show stats
name: collectd
tags: bind=127.0.0.1:25826
batches_tx      bytes_rx        points_rx       points_tx
----------      --------        ---------       ---------
7536            63940108        1814973         1814872


name: cq
--------
points_written  query_ok
254968          163050


name: engine
tags: path=/var/opt/influxdb/data/_internal/monitor/12, version=bz1
blks_write      blks_write_bytes        blks_write_bytes_c      points_write    points_write_dedupe
----------      ----------------        ------------------      ------------    -------------------
67              218988                  85700                   3792            3792


name: engine
tags: path=/var/opt/influxdb/data/collectd/default/6, version=bz1
blks_write      blks_write_bytes        blks_write_bytes_c      points_write    points_write_dedupe
----------      ----------------        ------------------      ------------    -------------------
6699            26680437                12374996                1270497         1270497


name: engine
tags: path=/var/opt/influxdb/data/singular/singular/4, version=bz1
blks_write      blks_write_bytes        blks_write_bytes_c      points_write    points_write_dedupe
----------      ----------------        ------------------      ------------    -------------------
70096           82826851                32007388                388642          247805


name: engine
tags: path=/var/opt/influxdb/data/singular/singular_180d/13, version=bz1
blks_write      blks_write_bytes        blks_write_bytes_c      points_write    points_write_dedupe
----------      ----------------        ------------------      ------------    -------------------
21145           73634467                27209880                163638          163638


name: httpd
tags: bind=127.0.0.1:8086
ping_req        points_written_fail     points_written_ok       query_req       query_resp_bytes        req     write_req       write_req_bytes
--------        -------------------     -----------------       ---------       ----------------        ---     ---------       ---------------
10              2180364                 581192                  271             3397081                 1381059 1380778         2426711950


name: shard
tags: engine=bz1, id=12, path=/var/opt/influxdb/data/_internal/monitor/12
fields_create   series_create   write_points_ok write_req
-------------   -------------   --------------- ---------
3               25              195795          7857


name: shard
tags: engine=bz1, id=6, path=/var/opt/influxdb/data/collectd/default/6
fields_create   series_create   write_points_ok write_req
-------------   -------------   --------------- ---------
0               268             1814872         7536


name: shard
tags: engine=bz1, id=4, path=/var/opt/influxdb/data/singular/singular/4
fields_create   series_create   write_points_fail       write_points_ok write_req
-------------   -------------   -----------------       --------------- ---------
63              37523           1090182                 672522          1299850


name: shard
tags: engine=bz1, id=13, path=/var/opt/influxdb/data/singular/singular_180d/13
fields_create   series_create   write_points_ok write_req
-------------   -------------   --------------- ---------
326             3350            163638          81819


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/10
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/11
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/12
auto_flush      mem_size        meta_flush      points_write    points_write_req
----------      --------        ----------      ------------    ----------------
78559           9830799         130             195795          7857


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/3
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/5
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/7
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/8
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/_internal/monitor/9
auto_flush      meta_flush
----------      ----------
78559           130


name: wal
tags: path=/var/opt/influxdb/wal/collectd/default/6
auto_flush      flush_duration  mem_flush       mem_size        meta_flush      points_flush    points_write    points_write_req        series_flush
----------      --------------  ---------       --------        ----------      ------------    ------------    ----------------        ------------
78566           1.590861264     1               9880349         130             1233675         1814872         7536                    231


name: wal
tags: path=/var/opt/influxdb/wal/singular/singular/2
auto_flush      meta_flush
----------      ----------
78567           130


name: wal
tags: path=/var/opt/influxdb/wal/singular/singular/4
auto_flush      flush_duration  mem_flush       mem_size        meta_flush      points_flush    points_write    points_write_req        series_flush    write_fail
----------      --------------  ---------       --------        ----------      ------------    ------------    ----------------        ------------    ----------
8321            3.074472785     5               104858206       14              466347          2852886         1299850                 65715           1090182


name: wal
tags: path=/var/opt/influxdb/wal/singular/singular_180d/13
auto_flush      flush_duration          idle_flush      mem_flush       mem_size        meta_flush      points_flush    points_write    points_write_req        series_flush
----------      --------------          ----------      ---------       --------        ----------      ------------    ------------    ----------------        ------------
77200           1.1577203900000002      1               3               0               128             163638          163638          81819                   5996


name: write
-----------
point_req       point_req_local req     write_error     write_ok
5027191         5027191         1397062 1090182         306880


name: runtime
-------------
Alloc           Frees           HeapAlloc       HeapIdle        HeapInUse       HeapObjects     HeapReleased    HeapSys         Lookups Mallocs         NumGC   NumGoroutine    PauseTotalNs    Sys             TotalAlloc
1304893184      3306202897      1304893184      1793540096      1583431680      8807989         1266188288      3376971776      1141    3315010886      2494    182             31168472627     3530562496      1886243527696






> show shards
name: collectd
--------------
id      start_time              end_time                expiry_time             owners
6       2015-09-14T00:00:00Z    2015-09-21T00:00:00Z    2015-09-21T00:00:00Z    1


name: singular
--------------
id      start_time              end_time                expiry_time             owners
2       2015-09-07T00:00:00Z    2015-09-14T00:00:00Z    2016-09-13T00:00:00Z    1
4       2015-09-14T00:00:00Z    2015-09-21T00:00:00Z    2016-09-20T00:00:00Z    1
13      2015-09-14T00:00:00Z    2015-09-21T00:00:00Z    2016-03-19T00:00:00Z    1


name: _internal
---------------
id      start_time              end_time                expiry_time             owners
3       2015-09-13T00:00:00Z    2015-09-14T00:00:00Z    2015-09-21T00:00:00Z    1
5       2015-09-14T00:00:00Z    2015-09-15T00:00:00Z    2015-09-22T00:00:00Z    1
7       2015-09-15T00:00:00Z    2015-09-16T00:00:00Z    2015-09-23T00:00:00Z    1
8       2015-09-16T00:00:00Z    2015-09-17T00:00:00Z    2015-09-24T00:00:00Z    1
9       2015-09-17T00:00:00Z    2015-09-18T00:00:00Z    2015-09-25T00:00:00Z    1
10      2015-09-18T00:00:00Z    2015-09-19T00:00:00Z    2015-09-26T00:00:00Z    1
11      2015-09-19T00:00:00Z    2015-09-20T00:00:00Z    2015-09-27T00:00:00Z    1
12      2015-09-20T00:00:00Z    2015-09-21T00:00:00Z    2015-09-28T00:00:00Z    1
```

config file:

```
[meta]
  dir = "/var/opt/influxdb/meta"
  hostname = "localhost"
  bind-address = "127.0.0.1:8088"
  retention-autocreate = true
  election-timeout = "1s"
  heartbeat-timeout = "1s"
  leader-lease-timeout = "500ms"
  commit-timeout = "50ms"
  cluster-tracing = false

[data]
  dir = "/var/opt/influxdb/data"
  max-wal-size = 104857600
  wal-flush-interval = "10m"
  wal-partition-flush-delay = "2s"
  wal-dir = "/var/opt/influxdb/wal"
  wal-logging-enabled = true

  wal-ready-series-size = 25600
  wal-compaction-threshold = 0.6
  wal-max-series-size = 2097152
  wal-flush-cold-interval = "10m"
  wal-partition-size-threshold = 20971520

[cluster]
  force-remote-mapping = false
  write-timeout = "5s"
  shard-writer-timeout = "5s"
  shard-mapper-timeout = "5s"

[retention]
  enabled = true
  check-interval = "30m0s"

[shard-precreation]
  enabled = true
  check-interval = "10m0s"
  advance-period = "30m0s"

[admin]
  enabled = false
  bind-address = "127.0.0.1:8083"
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"

[monitor]
  store-enabled = true
  store-database = "_internal"
  store-retention-policy = "monitor"
  store-retention-duration = "168h0m0s"
  store-replication-factor = 1
  store-interval = "10s"

[http]
  enabled = true
  bind-address = "127.0.0.1:8086"
  auth-enabled = false
  log-enabled = false
  write-tracing = false
  pprof-enabled = false
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"

[collectd]
  enabled = true
  bind-address = "127.0.0.1:25826"
  database = "collectd"
  retention-policy = ""
  batch-size = 1000
  batch-pending = 5
  batch-timeout = "10s"
  typesdb = "/usr/share/collectd/types.db"

[opentsdb]
  enabled = false
  bind-address = ":4242"
  database = "opentsdb"
  retention-policy = ""
  consistency-level = "one"
  tls-enabled = false
  certificate = "/etc/ssl/influxdb.pem"
  batch-size = 1000
  batch-pending = 5
  batch-timeout = "1s"

[continuous_queries]
  log-enabled = true
  enabled = true
  recompute-previous-n = 2
  recompute-no-older-than = "10m"
  compute-runs-per-interval = 10
  compute-no-more-than = "2m"

[hinted-handoff]
  enabled = false
  dir = "/var/opt/influxdb/hh"
  max-size = 1073741824
  max-age = "168h0m0s"
  retry-rate-limit = 0
  retry-interval = "1s"
```

**Comment by brunohass** [comment]
More logs..

From yesterday:

```
[wal] 2015/09/18 23:42:21 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:42:21 write to index of partition 1 took 2.863613ms
[wal] 2015/09/18 23:42:31 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:42:31 write to index of partition 1 took 2.57621ms
[wal] 2015/09/18 23:42:41 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:42:41 write to index of partition 1 took 2.494716ms
[wal] 2015/09/18 23:42:51 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:42:51 write to index of partition 1 took 2.77843ms
[wal] 2015/09/18 23:43:01 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:43:01 write to index of partition 1 took 2.693037ms
[wal] 2015/09/18 23:43:11 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:43:11 write to index of partition 1 took 2.141891ms
[wal] 2015/09/18 23:43:20 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:43:20 write to index of partition 1 took 2.398809ms
[wal] 2015/09/18 23:43:30 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:43:30 write to index of partition 1 took 2.353349ms
[wal] 2015/09/18 23:43:40 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:43:40 write to index of partition 1 took 2.377913ms
[wal] 2015/09/18 23:43:50 Flush due to idle. Flushing 27 series with 27 points and 1287 bytes from partition 1
[wal] 2015/09/18 23:43:50 write to index of partition 1 took 2.600593ms
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:50 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/18 23:43:51 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
```


From today:

```
[wal] 2015/09/20 00:07:29 Flush due to memory. Flushing 12326 series with 77705 points and 20972974 bytes from partition 1
[retention] 2015/09/20 00:07:32 retention policy enforcement check commencing
[retention] 2015/09/20 00:07:32 retention policy shard deletion check commencing
[retention] 2015/09/20 00:37:32 retention policy shard deletion check commencing
[retention] 2015/09/20 00:37:32 retention policy enforcement check commencing
[retention] 2015/09/20 01:07:32 retention policy shard deletion check commencing
[retention] 2015/09/20 01:07:32 retention policy enforcement check commencing
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
[write] 2015/09/20 01:18:54 write failed for shard 4 on node 1: engine: write points: write throughput too high. backoff and retry
```

**Comment by jwilder** [comment]
There is an issue with the current storage engines where the WAL can accept writes faster than it can flush them to the bolt index.  Previously, this would cause the server to run out of memory and panic.  0.9.4 will stop accepting writes when it's overloaded.  #4086 is being worked on to address this issue.

**Comment by mtzaurus** [comment]
May I ask when this issue is gonna be fixed ?
I am using that release:

         0.9.4	"da1b59e7d7764d36786253b2db13b97f42ed4e1d"	"0.9.4.2"

And it's still there.

**Comment by otoolep** [comment]
This issue is being fixed by creating a new engine. If you want to try the new engine, you can install a nightly build and set engine selection to `tsm1`, but you should be aware you will need to delete all data between upgrades until 0.9.5 is officially released. `tsm1` is still experimental.

**Comment by mtzaurus** [comment]
Engine is changed like this ?

[data]
  engine = "tsm1"

**Comment by otoolep** [comment]
Yes @mtzaurus

**Comment by otoolep** [comment]
You will not have new data written into the new engine for a while, depending on your retention period. Create a database if you want to force this to take place, and write data into that database.

**Comment by RobWin** [comment]
+1
Same issue here.

**Comment by vaizki** [comment]
+1 
Same here, just noticed no data was going in and I got it with quite low write rates.

Running InfluxDB v0.9.4.2 (git: 0.9.4 da1b59e7d7764d36786253b2db13b97f42ed4e1d)

Writes:
* +3896 points written every 15 minutes (in one HTTP POST)
* +1 point written every minute
* has been collecting for only 3 days now
* writes are always to existing series 

Reads are very low (one grafana dashboard with 4 graphs)

Restart fixed it. when 0.9.5 comes out, will there be a tool to migrate data to the new engine automatically? Is there something in _internal that can be queried to detect this situation?

**Comment by beckettsean** [comment]
@vaizki 

> when 0.9.5 comes out, will there be a tool to migrate data to the new engine automatically? 

When the `tsm1` engine is official there will be a tool to migrate existing shards to the `tsm1` engine. InfluxDB 0.9.5 has been released, but `tsm1` is still under development.

You should upgrade to 0.9.5.1. We found a deadlock between the write and query paths that might have led to the issues you are seeing.

**Comment by vaizki** [comment]
Thanks Sean, I did manage to notice (barely!) that 0.9.5 actually was released (apparently a blog post was made but then removed?) and upgraded to 0.9.5.1 a few days ago. No problems with the upgrade at all. 

An additional point I forgot to mention is that I write a lot into the future. This is because I am fetching weather data including forecasts 36 hours ahead and pushing them into influxdb. This is probably not a very common use case and might bring out issues.

**Comment by beckettsean** [comment]
@vaizki writing points into the future is definitely not a typical use case, and it might lead to decreased performance. I haven't done too much testing on it. If you see problems other than this one please open a new issue for them.

**Comment by xiongshuai** [comment]
Hi, dear beckettsean. I have found the issue "write points: write throughput too high. backoff and retry" in 0.9.5.1,  and sometimes the process of influxdb almost exhaust the memory of server. how to solve this problem, please help me.

**Comment by xiongshuai** [comment]
Is the issue caused by this source code? Can you please explain it? 
if shouldFailWrite, shouldCompact := func() (shouldFailWrite bool, shouldCompact bool) {
		p.mu.RLock()
		defer p.mu.RUnlock()
		// Return an error if memory threshold has been reached.
		if p.memorySize > p.sizeThreshold {
			if !p.compactionRunning {
				shouldCompact = true
			} else if p.memorySize > p.sizeThreshold*FailWriteMemoryThreshold {
				shouldFailWrite = true
			}
		}
		return
	}(); shouldCompact {
		go p.flushAndCompact(memoryFlush)
	} else if shouldFailWrite {
		p.statMap.Add(statWriteFail, 1)
		return fmt.Errorf("write throughput too high. backoff and retry")
	}

**Comment by beckettsean** [comment]
@xiongshuai you are writing too many points for the available resources and the server is telling you to slow down the rate. Add RAM and CPU or write fewer points per second.

**Comment by xiongshuai** [comment]
We write realtime log data. Sometimes, there are many points per second, but the number of points per second is very small from 0:00am to 9:00am everyday.  When our process is running for a few days, this issue occurs, and points can't be written to influxdb for several days, it looks like memory can't be free all the time. But one day the influxdb is back to normal and points can be written.

**Comment by beckettsean** [comment]
@xiongshuai please give more precise numbers on throughput (what is "many"? what is "very small"?) and provide some relevant log output. There isn't enough detail to diagnose anything.

> points can't be written to influxdb for several days

OOM killer should bounce the process. Does it not recover after a restart?

> it looks like memory can't be free all the time

I don't understand what you mean by this. Can you restate it with some data?

**Comment by xiongshuai** [comment]
I found that grafana consume a lot of memory in influxdb when it represent graph for visualizing time series data. With grafana, when I queried data for several days and several thousands points per minute, CPU and memory was almost exhausted,  at this time, points couldn't written to influxdb for timeout. Then I closed the page for grafana, CPU and memory slowed down,  but points couldn't written to influxdb for a very long time. 

   PID   USER      PR  NI    VIRT  RES  SHR   S %CPU %MEM    TIME+  COMMAND                                                                                                      
 79367 influxdb   20   0      26.1g  6.2g   995m S  1.0       20.0      169:47.66 influxd

[root@localhost:influxdb]# tail influxd.log
[http] 2016/01/18 11:18:00 10.100.2.7 - root [18/Jan/2016:11:17:55 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 1149326d-bd92-11e5-a2aa-000000000000 5.000651009s
[http] 2016/01/18 11:18:00 10.100.2.5 - root [18/Jan/2016:11:17:55 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 11538554-bd92-11e5-a2ab-000000000000 5.000626405s
[http] 2016/01/18 11:18:01 10.100.2.54 - root [18/Jan/2016:11:17:56 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 11782b93-bd92-11e5-a2ac-000000000000 5.000509112s
[http] 2016/01/18 11:18:01 10.100.2.51 - root [18/Jan/2016:11:17:56 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 1183ed32-bd92-11e5-a2ad-000000000000 5.000674483s
[http] 2016/01/18 11:18:02 10.100.2.53 - root [18/Jan/2016:11:17:57 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 124bb8f0-bd92-11e5-a2ae-000000000000 5.000643313s
[http] 2016/01/18 11:18:04 10.100.2.123 - root [18/Jan/2016:11:17:59 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 135b7150-bd92-11e5-a2af-000000000000 5.000695994s
[monitor] 2016/01/18 11:18:04 failed to store statistics: timeout
[http] 2016/01/18 11:18:04 10.100.2.83 - root [18/Jan/2016:11:17:59 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 137d07a3-bd92-11e5-a2b0-000000000000 5.000697239s
[http] 2016/01/18 11:18:04 10.100.2.123 - root [18/Jan/2016:11:17:59 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 138f2cba-bd92-11e5-a2b1-000000000000 5.00189424s
[http] 2016/01/18 11:18:04 10.100.2.84 - root [18/Jan/2016:11:17:59 +0800] POST /write?consistency=&db=realtime&precision=s&rp= HTTP/1.1 500 32 - InfluxDBClient 1392e394-bd92-11e5-a2b2-000000000000 5.000586711s

[root@localhost:influxdb]# grep -e "timeout" influxd.log|grep -e "2016/01/18 10:30" |more
[monitor] 2016/01/18 10:30:34 failed to store statistics: timeout
[monitor] 2016/01/18 10:30:44 failed to store statistics: timeout
[monitor] 2016/01/18 10:30:56 failed to store statistics: timeout

[root@localhost:influxdb]# grep -e "timeout" influxd.log|grep -e "2016/01/18 11:11" |more
[monitor] 2016/01/18 11:11:04 failed to store statistics: timeout
[monitor] 2016/01/18 11:11:14 failed to store statistics: timeout
[monitor] 2016/01/18 11:11:24 failed to store statistics: timeout
[monitor] 2016/01/18 11:11:34 failed to store statistics: timeout
[monitor] 2016/01/18 11:11:44 failed to store statistics: timeout
[monitor] 2016/01/18 11:11:54 failed to store statistics: timeout

**Comment by xiongshuai** [comment]
I used grafana about 1 hour ago which resulted in this issue, then I closed the page, server did not recover by now

**Comment by xiongshuai** [comment]
I used grafana about 1 hour ago which resulted in this issue, then I closed the page, influxdb did not recover by now

**Comment by xiongshuai** [comment]
So once the throughput of query or write exceed the limit, even it is slowed down later, influxdb could not recover all the time. Only restart influxdb server can make it be normal.

**Comment by undera** [comment]
+1, I've also seen this. I had to fall back to MySQL because of this, since MySQL is able to not consume a lot of memory. Interesting story BTW - my http://loadosophia.org service migration attempt. I'd love to use InfluxDB, but memory consumption and query speed compared to MySQL made me fall back.

**Comment by pauldix** [comment]
This is fixed in 0.10
